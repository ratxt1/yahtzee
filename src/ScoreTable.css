.ScoreTable-section {
  display: inline-block;
  vertical-align: top;
  width: 10em;
}

.ScoreTable-section-lower {
  margin-left: 3em;
}

.ScoreTable-total {
  margin-top: 2em;
}