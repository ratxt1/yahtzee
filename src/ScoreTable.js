import React, { Component } from 'react';
import './ScoreTable.css';
import {
  ones,
  twos,
  threes,
  fours,
  fives,
  sixes,
  threeOfKind,
  fourOfKind,
  fullHouse,
  smallStraight,
  largeStraight,
  yahtzee,
  chance
} from './Rules';
import RuleColumn from './RuleColumn';


class ScoreTable extends Component {

  render() {
    const { scores, doScore } = this.props;

    const upper = {
      name: "Upper",
      rows: [
        { name: "Ones", score: scores.ones, doScore: () => doScore("ones", ones.evalRoll) },
        { name: "Twos", score: scores.twos, doScore: () => doScore("twos", twos.evalRoll) },
        { name: "Threes", score: scores.threes, doScore: () => doScore("threes", threes.evalRoll) },
        { name: "Fours", score: scores.fours, doScore: () => doScore("fours", fours.evalRoll) },
        { name: "Fives", score: scores.fives, doScore: () => doScore("fives", fives.evalRoll) },
        { name: "Sixes", score: scores.sixes, doScore: () => doScore("sixes", sixes.evalRoll) },
      ],
      diceRolling: this.props.diceRolling
    }

    const lower = {
      name: "Lower",
      rows: [
        { name: "Three of a Kind", score: scores.threeOfKind, doScore: () => doScore("threeOfKind", threeOfKind.evalRoll) },
        { name: "Four of a Kind", score: scores.fourOfKind, doScore: () => doScore("fourOfKind", fourOfKind.evalRoll) },
        { name: "Full House", score: scores.fullHouse, doScore: () => doScore("fullHouse", fullHouse.evalRoll) },
        { name: "Small Straight", score: scores.smallStraight, doScore: () => doScore("smallStraight", smallStraight.evalRoll) },
        { name: "Large Straight", score: scores.largeStraight, doScore: () => doScore("largeStraight", largeStraight.evalRoll) },
        { name: "Yahtzee", score: scores.yahtzee, doScore: () => doScore("yahtzee", yahtzee.evalRoll) },
        { name: "Chance", score: scores.chance, doScore: () => doScore("chance", chance.evalRoll) },
      ],
      diceRolling: this.props.diceRolling
    }

    return (
      <div className="ScoreTable">
        <section className="ScoreTable-section">
          <RuleColumn {...upper} />
        </section>
        <section className="ScoreTable-section ScoreTable-section-lower">
          <RuleColumn {...lower} />
        </section>
        <h3 className="ScoreTable-total"> Total Score: {
          Object.values(scores).reduce((acc, val) => {
            if (val !== undefined) {
              acc += val
            }
            return acc
          }, 0)
        }
        </h3>


      </div>
    )
  }
}

export default ScoreTable;
