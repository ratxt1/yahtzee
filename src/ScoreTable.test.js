import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import ScoreTable from "./ScoreTable";


it("renders without crashing", function() {
  shallow(<ScoreTable doScore={() => {}} scores={{ones: undefined}} />);
});

it("matches snapshot", function() {
  let wrapper = shallow(<ScoreTable doScore={() => {}} scores={{ones: undefined}} />);
  let serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});