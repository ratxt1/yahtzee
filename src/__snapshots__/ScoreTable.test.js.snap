// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="ScoreTable"
>
  <section
    className="ScoreTable-section"
  >
    <RuleColumn
      name="Upper"
      rows={
        Array [
          Object {
            "doScore": [Function],
            "name": "Ones",
            "score": undefined,
          },
          Object {
            "doScore": [Function],
            "name": "Twos",
            "score": undefined,
          },
          Object {
            "doScore": [Function],
            "name": "Threes",
            "score": undefined,
          },
          Object {
            "doScore": [Function],
            "name": "Fours",
            "score": undefined,
          },
          Object {
            "doScore": [Function],
            "name": "Fives",
            "score": undefined,
          },
          Object {
            "doScore": [Function],
            "name": "Sixes",
            "score": undefined,
          },
        ]
      }
    />
  </section>
  <section
    className="ScoreTable-section ScoreTable-section-lower"
  >
    <RuleColumn
      name="Lower"
      rows={
        Array [
          Object {
            "doScore": [Function],
            "name": "Three of a Kind",
            "score": undefined,
          },
          Object {
            "doScore": [Function],
            "name": "Four of a Kind",
            "score": undefined,
          },
          Object {
            "doScore": [Function],
            "name": "Full House",
            "score": undefined,
          },
          Object {
            "doScore": [Function],
            "name": "Small Straight",
            "score": undefined,
          },
          Object {
            "doScore": [Function],
            "name": "Large Straight",
            "score": undefined,
          },
          Object {
            "doScore": [Function],
            "name": "Yahtzee",
            "score": undefined,
          },
          Object {
            "doScore": [Function],
            "name": "Chance",
            "score": undefined,
          },
        ]
      }
    />
  </section>
  <h3
    className="ScoreTable-total"
  >
     Total Score: 
    0
  </h3>
</div>
`;
